src/App.tsx

```diff
@@ -15,48 +17,115 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  default = '',
+  name = 'name',
+  length = 'length',
+}
+
+interface FilterParams {
+  sortField: SortType;
+  isReversed: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField, isReversed }: FilterParams,
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.length:
+          return good1.length - good2.length;
+
+        case SortType.name:
+          return good1.localeCompare(good2);
+
+        default:
+          return 0;
+      }
```

Comment:
you have only 2 conditions, i think there is no necessary use `switch { case }` (not critical)